src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt

```diff
@@ -363,18 +350,44 @@ class CradleMessageExtractor(
     private inline fun <T> measure(name: String, action: () -> T): T = dataMeasurement.start(name).use { action() }
 }
 
+internal class GroupBatchCheckIterator(
+    private val original: Iterator<StoredGroupedMessageBatch>,
+) : Iterator<StoredGroupedMessageBatch> by original {
+    override fun next(): StoredGroupedMessageBatch = original.next().also { batch ->
+        if (batch.messages.size > 1) {
+            batch.messages.asSequence().chunked(2).forEach { pair ->
```

Comment:
You can use destructive declaration here
```suggestion
            batch.messages.asSequence().chunked(2).forEach { (first, second) ->
```